teuthology/orchestra/remote.py

```diff
@@ -535,22 +537,25 @@ def read_file(self, path, sudo=False, stdout=None,
         :returns: the file contents in bytes, if stdout is `io.BytesIO`, by default
         :returns: the file contents in str, if stdout is `io.StringIO`
         """
-        dd = 'sudo dd' if sudo else 'dd'
-        args = dd + ' if=' + path + ' of=/dev/stdout'
-        iflags=[]
+        def maybe_sudo(cmd):
+            return 'sudo {cmd}' if sudo else cmd
```

Comment:
Interesting idea, but this does not work as expected? Can we revert it here and later?

Reply:
fixed and repushed.